eventually/vote/views.py

```diff
@@ -0,0 +1,337 @@
+
+        :param vote_id: id of the certain vote
+        :type vote_id: int
+
+        :return: HttpResponse with status 400,404 or 201
+        """
+
+        user = request.user
+        team = Team.get_by_id(team_id)
+
+        data = request.body
+        if not data:
+            return HttpResponse(status=400)
+
+        if not vote_id:
+            return HttpResponse(status=400)
+
+        members = [user.id for user in team.members.all()]
+        if user.id not in members:
+            return HttpResponse(status=400)
+
+        vote = Vote.get_by_id(vote_id)
+        if not vote:
+            return HttpResponse(status=400)
+
+        if data.get('title') and string_validator(data['title']):
+            set_title = data.get('title')
+
+        data = {
+            'title': set_title,
+            'is_active': data.get("is_active"),
+            'is_extended': data.get("is_extended"),
+            'vote_type': data.get("vote_type")
+        }
+        vote.update(**data)
+        vote.to_dict()
+        return HttpResponse(vote, status=200)
+
+    def delete(self, request, team_id, event_id, vote_id=None):  # pylint: disable=unused-argument
+        """
+        Method that handles DELETE request.
+
+        :param request: the accepted HTTP request
+        :type request: HttpRequest object
+
+        :param team_id: id of the certain team
+        :type team_id: int
+
+        :param event_id: id of the certain event
+        :type event_id: int
+
+        :param vote_id: id of the certain vote
+        :type vote_id: int
+
+        :return: HttpResponse with status 400,404 or 200
+        """
+        user = request.user
+        team = Team.get_by_id(team_id)
+
+        if not vote_id:
+            return HttpResponse(status=400)
+
+        members = [user.id for user in team.members.all()]
+        if user.id not in members:
+            return HttpResponse(status=400)
+
+        vote = Vote.get_by_id(vote_id)
+        if not vote:
+            return HttpResponse(status=404)
+
+        Vote.delete_by_id(vote_id)
+        return HttpResponse(status=200)
+
+
+class AnswerView(View):
+    """VoteView handles GET, POST, PUT, DELETE requests"""
+
+    def get(self, request, team_id=None, event_id=None, vote_id=None,
+            answer_id=None):  # pylint: disable=unused-argument
+        """
+        Method that handles GET request.
+
+        :param request: the accepted HTTP request
+        :type request: HttpRequest object
+
+        :param team_id: id of the certain team
+        :type team_id: int
+
+        :param event_id: id of the certain event
+        :type event_id: int
+
+        :param vote_id: id of the certain vote
+        :type vote_id: int
+
+        :param answer_id: id of the certain answer
+        :type answer_id: int
+
+        :return: HttpResponse with status 400,404 or 200
+        """
+        user = request.user
+        team = Team.get_by_id(team_id)
+
+        if not answer_id:
```

Comment:
maybe we need some logic in the case when we do not receive the answer_id? When we want to see all answers. How do you think?

Reply:
+